module #4. Authentication & Spring Security/oauth2-resource-server/src/main/java/com/epam/esm/controller/GiftCertificateController.java

```diff
@@ -83,6 +83,7 @@ public GiftCertificateDto getById(@PathVariable @Valid @Min(value = 1, message =
      * @param giftCertificateDto the {@link GiftCertificateDto Gift Certificate} to be saved
      * @return saved {@link GiftCertificateDto Gift Certificate}
      */
+    @PreAuthorize(value = "hasRole('ADMIN') and hasAuthority('SCOPE_certificate.write')")
```

Comment:
Maybe you wanted to write hasRoleAdmin OR hasAuthority... istead of AND?

Reply:
Because you could have technical users which shouldn`t be admins but could have appropriate scope